x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/token/CreateTokenResponse.java

```diff
@@ -93,6 +105,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (kerberosAuthenticationResponseToken != null) {
             builder.field("kerberos_authentication_response_token", kerberosAuthenticationResponseToken);
         }
+        builder.field("authentication", authentication);
```

Comment:
I think we need a null check here to not show the field at all if the value is `null`. As far as I understand it, this is the general preference (see other fields in this method). This comment applies to other places where xcontent is built with authentication field.

Reply:
I understand that "scope" and "kerberos_authentication_response_token" may not be null depending on the type of token we create, but authentication object should always exist (if we succeed), see "access_token" and "expires_in" fields in the method.

Reply:
It can be `null` if the object is [deserialised](https://github.com/elastic/elasticsearch/pull/62490/files#diff-505d2ad6a63d3db527aa6c7aeee9e4fa7586c8c17a0651774ddc75b279c67fcbR40-R45) from bytes created prior to `v7.11`. We have the `null` check for it in both `CreateTokenResponse(StreamInput in)` and `CreateTokenResponse#writeTo`, it makes sense to have it here as well.

Reply:
Make sense

Reply:
Done & done